packages/web-components/src/components/ic-side-navigation/ic-side-navigation.tsx

```diff
@@ -351,7 +356,7 @@ export class SideNavigation {
       secondaryNavigationWrapper.classList.add(collapsedIconLabelsStart);
     }
 
-    this.el.addEventListener("transitionend", () => {
+    const transitionEndHandler = () => {
```

Comment:
again, i think this handler needs to be moved outside of the scope of the `animateCollapsedIconLabels `function, as it will be creating a new instance of `transitionEndHandler`, every time `animateCollapsedIconLabels ` is called